webviz_subsurface_components/VectorCalculatorWrapper.py

```diff
@@ -0,0 +1,204 @@
+from typing import List, Dict, Union
+from functools import wraps
+import sys
+import re
+import warnings
+
+if sys.version_info[0] == 3 and sys.version_info[1] >= 8:
+    from typing import TypedDict
+else:
+    from typing_extensions import TypedDict
+
+import numpy as np
+from py_expression_eval import Parser
+
+from .VectorCalculator import VectorCalculator
+
+
+class VariableVectorMapInfo(TypedDict):
+    variableName: str
+    vectorName: List[str]
+
+
+class ExpressionInfo(TypedDict):
+    name: str
+    expression: str
+    id: str
+    variableVectorMap: List[VariableVectorMapInfo]
+    isValid: bool
+    isDeletable: bool
+
+
+class ExternalParseData(TypedDict):
+    expression: str
+    id: str
+    variables: List[str]
+    isValid: bool
+    message: str
+
+
+class VectorCalculatorParser(Parser):
+    """Creates expression parser configured to handle vector variables
+
+    Overrides operators and functions to provide whitelist for parsing. Note that the overrides also replace
+    math lib functions with numpy functions for handling array data.
+
+    I.e.: Configured expression parser for vector calculator
+    """
+
+    def __init__(self) -> None:
+        super(VectorCalculatorParser, self).__init__()
+
+        self.characterBlacklist = ['"', "'"]
+
+        # Whitelist with numpy operators
+        self.ops1 = {
+            "sqrt": np.sqrt,
+            "abs": np.abs,
+            "-": np.negative,
+        }
+        # Whitelist with numpy operators
+        self.ops2 = {
+            "+": np.add,
+            "-": np.subtract,
+            "*": np.multiply,
+            "/": np.divide,
+            "^": np.power,
+        }
+        # Whitelist with numpy functions
+        self.functions = {
+            "log": np.log,  # Natural logarithm
```

Comment:
```suggestion
            "ln": np.log,  # Natural logarithm
```

Reply:
Also added as "whitelisted" function in JS-parser